File: mobiledevelopment/lib/src/config/theme.dart
import 'package:flutter/material.dart';
import '../styles/theme.dart';

const themeName = 'Blue Diamond';

class StatGradients {
  static const a = LinearGradient(
    colors: [AppColors.brand100, AppColors.brand200],
    begin: Alignment.topLeft,
    end: Alignment.bottomRight,
  );
  static const b = LinearGradient(
    colors: [AppColors.brand200, Color(0xFF9BBFE8)],
    begin: Alignment.topLeft,
    end: Alignment.bottomRight,
  );
  static const c = LinearGradient(
    colors: [AppColors.diamond100, AppColors.diamond200],
    begin: Alignment.topLeft,
    end: Alignment.bottomRight,
  );
  static const d = LinearGradient(
    colors: [AppColors.brand50, AppColors.diamond100],
    begin: Alignment.topLeft,
    end: Alignment.bottomRight,
  );
}

class StatusBadgeStyle {
  final Color background;
  final Color foreground;
  const StatusBadgeStyle({required this.background, required this.foreground});
}

const statusBadges = {
  'active': StatusBadgeStyle(background: AppColors.brand100, foreground: AppColors.brand800),
  'inactive': StatusBadgeStyle(background: Color(0xFFF1F5F9), foreground: AppColors.slate500),
  'pending': StatusBadgeStyle(background: AppColors.diamond100, foreground: AppColors.diamond600),
  'approved': StatusBadgeStyle(background: AppColors.brand200, foreground: AppColors.brand900),
  'draft': StatusBadgeStyle(background: Color(0xFFF1F5F9), foreground: AppColors.slate500),
  'completed': StatusBadgeStyle(background: AppColors.brand100, foreground: AppColors.brand800),
  'rejected': StatusBadgeStyle(background: Color(0xFFFEE2E2), foreground: Color(0xFFB91C1C)),
  'success': StatusBadgeStyle(background: AppColors.brand100, foreground: AppColors.brand800),
  'warning': StatusBadgeStyle(background: AppColors.diamond100, foreground: AppColors.diamond600),
  'excellent': StatusBadgeStyle(background: AppColors.brand200, foreground: AppColors.brand900),
  'good': StatusBadgeStyle(background: AppColors.brand100, foreground: AppColors.brand700),
  'poor': StatusBadgeStyle(background: Color(0xFFFEE2E2), foreground: Color(0xFFB91C1C)),
};

StatusBadgeStyle statusBadgeStyle(String status) =>
    statusBadges[status] ?? StatusBadgeStyle(background: const Color(0xFFF1F5F9), foreground: AppColors.slate500);

Widget statusBadge(String status, {String? label}) {
  final style = statusBadgeStyle(status);
  return Container(
    padding: const EdgeInsets.symmetric(horizontal: 10, vertical: 5),
    decoration: BoxDecoration(
      color: style.background,
      borderRadius: BorderRadius.circular(999),
      border: Border.all(color: style.foreground.withValues(alpha: 0.15)),
    ),
    child: Text(
      label ?? status,
      style: TextStyle(color: style.foreground, fontSize: 11, fontWeight: FontWeight.w800, letterSpacing: 0.2),
    ),
  );
}

class ProgressColors {
  static const high = LinearGradient(colors: [AppColors.brand500, AppColors.diamond400]);
  static const medium = LinearGradient(colors: [AppColors.brand500, AppColors.brand600]);
  static const low = LinearGradient(colors: [AppColors.diamond400, AppColors.brand500]);
  static const danger = LinearGradient(colors: [Color(0xFFEF4444), Color(0xFFDC2626)]);
}

LinearGradient progressColor(double percent) {
  if (percent > 80) return ProgressColors.danger;
  if (percent > 50) return ProgressColors.medium;
  return ProgressColors.high;
}

const textPositive = AppColors.brand600;
const textNegative = Color(0xFFDC2626);

File: src/config/theme.ts
/** Blue Diamond Theme — centralized tokens */

export const THEME_NAME = 'Blue Diamond';

export const STAT_GRADIENTS = {
  a: 'bg-gradient-to-br from-brand-100 to-brand-200 text-brand-700',
  b: 'bg-gradient-to-br from-brand-200 to-brand-300 text-brand-800',
  c: 'bg-gradient-to-br from-diamond-100 to-diamond-200 text-diamond-700',
  d: 'bg-gradient-to-br from-brand-50 to-diamond-100 text-brand-600',
} as const;

export const STATUS_BADGES = {
  active: 'bg-brand-100 text-brand-800',
  inactive: 'bg-slate-100 text-slate-600',
  pending: 'bg-diamond-100 text-diamond-800',
  approved: 'bg-brand-200 text-brand-900',
  draft: 'bg-slate-100 text-slate-600',
  completed: 'bg-brand-100 text-brand-800',
  rejected: 'bg-red-100 text-red-800',
  success: 'bg-brand-100 text-brand-800',
  warning: 'bg-diamond-100 text-diamond-800',
  excellent: 'bg-brand-200 text-brand-900',
  good: 'bg-brand-100 text-brand-700',
  poor: 'bg-red-100 text-red-800',
} as const;

export const PROGRESS_COLORS = {
  high: 'bg-gradient-to-r from-brand-500 to-diamond-400',
  medium: 'bg-gradient-to-r from-brand-400 to-brand-500',
  low: 'bg-gradient-to-r from-diamond-400 to-brand-400',
  danger: 'bg-gradient-to-r from-red-500 to-red-600',
} as const;

export const TEXT_POSITIVE = 'text-brand-600';
export const TEXT_NEGATIVE = 'text-red-600';
